main/EDInitMod.F90

```diff
@@ -450,80 +463,149 @@ subroutine set_site_properties( nsites, sites,bc_in )
           sites(s)%ema_npp = -9999._r8
 
           if(hlm_use_fixed_biogeog.eq.itrue)then
-             ! MAPPING OF FATES PFTs on to HLM_PFTs
-             ! add up the area associated with each FATES PFT
-             ! where pft_areafrac is the area of land in each HLM PFT and (from surface dataset)
-             ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
-
-             do hlm_pft = 1,size( EDPftvarcon_inst%hlm_pft_map,2)
-                do fates_pft = 1,numpft ! loop round all fates pfts for all hlm pfts
-                   sites(s)%area_pft(fates_pft) = sites(s)%area_pft(fates_pft) + &
-                        EDPftvarcon_inst%hlm_pft_map(fates_pft,hlm_pft) * bc_in(s)%pft_areafrac(hlm_pft)
-                end do
-             end do !hlm_pft
 
-             do ft =  1,numpft
-                if(sites(s)%area_pft(ft).lt.0.01_r8.and.sites(s)%area_pft(ft).gt.0.0_r8)then
-                   if(debug) write(fates_log(),*)  'removing small pft patches',s,ft,sites(s)%area_pft(ft)
-                   sites(s)%area_pft(ft)=0.0_r8
-                   ! remove tiny patches to prevent numerical errors in terminate patches
+             use_fates_luh_if: if (hlm_use_luh .eq. itrue) then
+                ! MAPPING OF FATES PFTs on to HLM_PFTs with land use
+                ! add up the area associated with each FATES PFT
+                ! where pft_areafrac_lu is the area of land in each HLM PFT and land use type (from surface dataset)
+                ! hlm_pft_map is the area of that land in each FATES PFT (from param file)
+
+                ! First check for NaNs in bc_in(s)%pft_areafrac_lu. If so, make everything bare ground.
+                if ( .not. (any( isnan( bc_in(s)%pft_areafrac_lu (:,:) )) .or. isnan( bc_in(s)%baregroundfrac))) then
```

Comment:
I wonder if this isnan will give us trouble.  I wonder if it would be better to initialize at a known unknown, like the  [hlm_hio_ignore_val](https://github.com/NGEET/fates/blob/sci.1.76.3_api.35.1.0/main/FatesInterfaceTypesMod.F90#L77-L80) or something else?

Reply:
Just to update that I looked at a run that @glemieux did with the most recent code, and it doesn't show the same artifact as in the figure above, so whatever I did to cause that doesn't seem to be persistent.